helm/templates/deployment.yaml

```diff
@@ -33,10 +33,6 @@ spec:
             {{- toYaml .Values.securityContext | nindent 12 }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
-          # ports:
-          #   - name: http
-          #     containerPort: 80
-          #     protocol: TCP
           # livenessProbe:
           #   httpGet:
           #     path: /
```

Comment:
It would be good to have a liveness probe even though we don't have a web frontend. What would a reasonable liveness test be for this deployment?

Reply:
I've written up issue #51 for this.

I struggled with this quite a bit. There are a couple of non-HTTP ways to add a livenessProbe: 
1. opening a TCP port and using a tcp probe. 
2. use an `exec` probe (eg to monitor last update time on a file on the pod's filesystem.

However, the tricky part is knowing how we can tell if the app is responsive. 
* `ps -A | grep java` only tells us if the process is alive - and if it's not, the pod would die anyway, with or without the probe.
* starting a separate thread just to update liveness only means the thread is working - it doesn't mean the indexer is indexing.

Any ideas? Is there a method I could call in the app, that will not do anything destructive?

Reply:
Clients interact by queueing tasks in rmq. So maybe check that rmq is available, and solr is available? the actual index worker is a task processor and is not really listening. I guess you could check if there are 1 or more idw pods in running state?

Reply:
Will merge and continue this conversation in Issue #51